app/validators/form_password_validator.rb

```diff
@@ -22,6 +23,15 @@ def strong_password
     errors.add :password, :weak_password, i18n_variables
   end
 
+  def password_graphemes_length
+    return if errors.messages.present?
+    if password.grapheme_clusters.length < Devise.password_length.min
+      errors.add(:password, :too_short, count: Devise.password_length.min)
+    elsif password.grapheme_clusters.length > Devise.password_length.max
+      errors.add(:password, :too_long, count: Devise.password_length.max)
```

Comment:
almost 100% covered :-)

Reply:
I did try for this, but it turns out to be a bit difficult (maybe even not possible?). Since this change will mostly avoids undercounting "characters", the regular length check will probably catch any overages before this length check would.